docs/reference/upgrade.md

```diff
@@ -11,11 +11,54 @@ redirect_from: ../deploying-aws/upgrade.html
 # Upgrading lakeFS
 {: .no_toc }
 Upgrading lakeFS from a previous version usually just requires re-deploying with the latest image (or downloading the latest version if you're using the binary).
-In some cases, the database may require a migration - check whether the [release](https://github.com/treeverse/lakeFS/releases) you're upgrading to requires that.
+If you're upgrading, check whether the [release](https://github.com/treeverse/lakeFS/releases) requires a migration.
 
 
 ## When DB migrations are required
 
+### lakeFS <TBD.X.X> or greater (KV Migration)
+
+Starting with version <TBD.X.X>, lakeFS has transitioned from using a PostgreSQL based database implementation to a Key-Value datastore interface supporting
+multiple database implementations. More information can be found [here](link-to-lakefs-on-kv-documentation).  
+Users upgrading from a previous version of lakeFS must pass through the KV migration version (<TBD.X.X>) before upgrading to newer versions of lakeFS.
+
+#### Migration Steps
```

Comment:
Can we also add a recommendation to commit uncommitted objects to speed up the migration?